xc/device.ml

```diff
@@ -2087,6 +2094,34 @@ module Backend = struct
              qmp_write domid (Qmp.Command(None, Qmp.Xen_set_global_dirty_log false));
           )
 
+      let cmdline_of_info ~xs ~dm info restore domid =
```

Comment:
The comments from above about `take`, `mapi`, `List.length` apply here as well.

Reply:
mapi is not needed here because we use the device id.